packages/analytics-controller/src/AnalyticsController.ts

```diff
@@ -176,119 +191,150 @@ export class AnalyticsController extends BaseController<
 > {
   readonly #platformAdapter: AnalyticsPlatformAdapter;
 
+  readonly #anonymousEventsFeature: boolean;
+
+  #initialized: boolean;
+
   /**
    * Constructs an AnalyticsController instance.
    *
    * @param options - Controller options
-   * @param options.state - Initial controller state (defaults from getDefaultAnalyticsControllerState)
+   * @param options.state - Initial controller state. Must include a valid UUIDv4 `analyticsId`.
+   * Use `getDefaultAnalyticsControllerState()` for default opt-in preferences.
    * @param options.messenger - Messenger used to communicate with BaseController
    * @param options.platformAdapter - Platform adapter implementation for tracking
+   * @param options.anonymousEventsFeature - Whether the anonymous events feature is enabled
+   * @throws Error if state.analyticsId is missing or not a valid UUIDv4
+   * @remarks After construction, call {@link AnalyticsController.init} to complete initialization.
    */
   constructor({
-    state = {},
+    state,
     messenger,
     platformAdapter,
+    anonymousEventsFeature = false,
   }: AnalyticsControllerOptions) {
+
+    const initialState: AnalyticsControllerState = {
+      ...getDefaultAnalyticsControllerState(),
+      ...state,
+    };
+
+    validateAnalyticsControllerState(initialState);
+
     super({
       name: controllerName,
       metadata: analyticsControllerMetadata,
-      state: {
-        ...getDefaultAnalyticsControllerState(),
-        ...state,
-      },
+      state: initialState,
       messenger,
-    });
+    }); 
 
+    this.#anonymousEventsFeature = anonymousEventsFeature;
     this.#platformAdapter = platformAdapter;
+    this.#initialized = false;
 
     this.messenger.registerMethodActionHandlers(
       this,
       MESSENGER_EXPOSED_METHODS,
     );
 
     projectLogger('AnalyticsController initialized and ready', {
-      enabled: this.state.enabled,
+      enabled: analyticsControllerSelectors.selectEnabled(this.state),
       optedIn: this.state.optedIn,
       analyticsId: this.state.analyticsId,
     });
   }
 
   /**
-   * Track an analytics event.
+   * Initialize the controller by calling the platform adapter's onSetupCompleted lifecycle hook.
+   * This method must be called after construction to complete the setup process.
    *
-   * Events are only tracked if analytics is enabled.
-   *
-   * @param eventName - The name of the event
-   * @param properties - Event properties
+   * @throws Error if called multiple times
    */
-  trackEvent(
-    eventName: string,
-    properties: AnalyticsEventProperties = {},
-  ): void {
-    // Don't track if analytics is disabled
-    if (!this.state.enabled) {
-      return;
+  init(): void {
+    if (this.#initialized) {
+      throw new Error('AnalyticsController already initialized.');
```

Comment:
Nit: Maybe we don't need to throw an error? Maybe we do nothing and use `console.warn` or `projectLogger` to print a warning instead?